### PR TITLE
Add build-time configuration of the vcs/ssh user

### PR DESCRIPTION
**Pull request by neomantra**
We use Mercurial as our primary DSCM.  It was awkward ssh'ing our `hg` repos using `git@server`.    This changeset makes this user configurable at container build-time.

If you like this, I can add some docs to the Advanced Configuration...

### CODE REVIEW

**Comment by hach-que** [comment]
Why does this do a `__PHABRICATOR_VCS_USER__` replace instead of just writing the setting into `/config.saved`?

**Comment by neomantra** [comment]
I missed `config.saved`, thanks.  The `PHABRICATOR_VCS_USER` should definitely be saved there.   I fixed it up so that all the init files use the environment variable.  However, the config files still need templating; I was on the fence about  `phabricator-ssh-hook.sh` and left it templated.

File: setup.sh [hunk 1]
```diff
@@ -52,20 +52,26 @@ rm setup.sh
 cd /
 rmdir /preflight # This should now be empty; it's an error if it's not.
 
+# Update the PHABRICATOR_VCS_USER in config files
+# NOTE: this variable must be set a container build-time, not at run-time!
```
Comment (hach-que) [inline setup.sh]: Remove note.

File: ENV-LIST.md [hunk 2]
```diff
@@ -22,6 +22,10 @@
 - `SSL_PRIVATE_KEY` - The path to the SSL private key (manual mode only).
 - `SSL_EMAIL` - The email address to use when registering for an SSL certificate (Let's Encrypt mode only)
 
+The following environment variable may be used at run-time, but must only be specified at build time using `--build-arg`:
```
Comment (hach-que) [inline ENV-LIST.md]: Doesn't need to be set at build time now?

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,8 @@
 FROM hachque/systemd-none
 
+ARG PHABRICATOR_VCS_USER
```
Comment (hach-que) [inline Dockerfile]: Can remove both of these since they're coming through `/config.saved`.

File: setup.sh [hunk 4]
```diff
@@ -52,20 +52,26 @@ rm setup.sh
...
+sed -i "s/__PHABRICATOR_VCS_USER__/$PHABRICATOR_VCS_USER/g" /etc/phabricator-ssh/sshd_config.phabricator
```
Comment (hach-que) [inline setup.sh]: These should go in the boot script after the configuration is initialised.

**Comment by hach-que** [comment]
I added some notes; basically I want to get rid of the build time configuration dependency entirely because I don't want yet another way of configuring the template (and one that requires you to build the image just to change this option).

**Comment by neomantra** [comment]
Thanks for the pronto review.  As currently structured, doesn't it have to be a build-time configuration?   `setup.sh` is run only at build-time, creates a user with `PHABRICATOR_VCS_USER` and builds a directory tree as that user.

Perhaps the phabricator cloning can happen as root and then in 10-boot, the user is created and the tree is chown'd ?

**Comment by hach-que** [comment]
Oh right.  Maybe let everything run through setup.sh as it currently does (maybe rename the user to `phabricator`), and then at the start of boot, run sed over `/etc/passwd` to rename the user?

**Comment by neomantra** [comment]
Oh good point, as long as the uid stays 2000, then it will all work.   Ok I will rework this with your comments. Thanks!

**Comment by hach-que** [comment]
Thanks

**Pull request merged**